frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -2017,6 +2076,54 @@ mod tests {
 		})
 	}
 
+	#[test]
+	fn accepts_good_solution_during_emergency() {
+		ExtBuilder::default().onchain_fallback(false).build_and_execute(|| {
+			roll_to(25);
+			assert_eq!(MultiPhase::current_phase(), Phase::Unsigned((true, 25)));
+
+			// No solutions are submitted so the queue should be empty
+			assert!(MultiPhase::queued_solution().is_none());
+			assert_eq!(MultiPhase::elect().unwrap_err(), ElectionError::Fallback("NoFallback."));
+
+			assert!(MultiPhase::current_phase().is_emergency());
+
+			let solution = crate::mock::raw_solution();
+			let origin = crate::mock::Origin::signed(99);
+
+			assert_ok!(MultiPhase::submit_emergency_solution(origin, Box::new(solution)));
+
+			// The queued solution shouldn't be none now because the submitted
+			// solution is correct.
+			assert!(MultiPhase::queued_solution().is_some());
```

Comment:
Can you assert the events that have happened here and check that they are sensible?